lib/appmap/service/test_command_provider.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'appmap/service/test_framework_detector'
+require 'appmap/service/integration_test_path_finder'
+
+module AppMap
+  module Service
+    class TestCommandProvider
+      class << self
+        def all
+          commands = []
+
+          if TestFrameworkDetector.rspec_present? && !integration_test_paths[:rspec].empty?
+            commands << {
+              framework: :rspec,
+              command: "APPMAP=true bundle exec rspec #{integration_test_paths[:rspec].join(' ')}"
+            }
+          end
+
+          if TestFrameworkDetector.minitest_present? && !integration_test_paths[:minitest].empty?
+            commands << {
+              framework: :minitest,
+              command: minitest_command
+            }
+          end
+
+          if TestFrameworkDetector.cucumber_present? && !integration_test_paths[:cucumber].empty?
+            commands << {
+              framework: :cucumber,
+              command: 'APPMAP=true bundle exec cucumber'
+            }
+          end
+
+          commands
+        end
+
+        private
+
+        def minitest_command
+          if Gem.loaded_specs.has_key?('rails')
+            "APPMAP=true bundle exec rails test #{integration_test_paths[:minitest].join(' ')}"
+          else
+            subcommands = integration_test_paths[:minitest].map { |path| "bundle exec ruby #{path}" }
+            "APPMAP=true #{subcommands.join(' && ')}"
```

Comment:
This will only set `APPMAP=true` for the first command:
```shell
bash-5.1$ FOO=bar bash -c 'echo first foo \"$FOO\"' && bash -c 'echo second foo \"$FOO\"'
first foo "bar"
second foo ""
bash-5.1$ FOO=first bash -c 'echo first foo \"$FOO\"' && FOO=second bash -c 'echo second foo \"$FOO\"'
first foo "first"
second foo "second"
```